controllers/mattermost/clusterinstallation/conversion.go

```diff
@@ -0,0 +1,169 @@
+package clusterinstallation
+
+import (
+	"context"
+
+	mattermostv1alpha1 "github.com/mattermost/mattermost-operator/apis/mattermost/v1alpha1"
+	mattermostv1beta1 "github.com/mattermost/mattermost-operator/apis/mattermost/v1beta1"
+	"github.com/mattermost/mattermost-operator/pkg/database"
+	"github.com/mattermost/mattermost-operator/pkg/utils"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *ClusterInstallationReconciler) IsConvertible(ci *mattermostv1alpha1.ClusterInstallation) error {
+	if ci.Spec.BlueGreen.Enable {
+		return errors.New("cluster installation with BlueGreen enabled cannot be converted to Mattermost")
```

Comment:
This might make it a bit clearer if someone runs into this in the logs

```suggestion
		return errors.New("ClusterInstallation resource with BlueGreen enabled cannot be converted to Mattermost resource. Disable BlueGreen to enable migration.")
```